### DISCUSSION TITLE
Releasing a glibc 2.28 friendly binary for elastix 5.3.0

### DISCUSSION THREAD

**Discussion by dorkylever** [SuperElastix/elastix#1082] [Q&A]
Hi All, 

Is it possible to release a linux binary for elastix 5.1.0 that's compatible with glibc 2.28? I'm presuming there's probably compilation issues with itk 5.3.0.

Kind Regards, 
Kyle

**Comment by N-Dekker** [comment]
Thanks for your question, Kyle. Can you please elaborate a little bit? Sorry, I'm not a linux expert. How does your use case look like, would you like to use the elastix library, or the elastix executable? Are you able to build elastix yourself, from the source?

Most interesting, of course, what are those compilation issues?

**Reply by dorkylever** [reply]
sorry, see post below.

**Comment by dorkylever** [comment]
Hi Neil, 

I'm a PhD student who is a current dev for [LAMA](https://github.com/dorkylever/LAMA).  We use command line calls via subprocessing in python scripts (I know it's not memory efficient).  We've been happily using the elastix 4.9.0 executable but want to update elastix for both the improved runtime as well as the in-built jacobian stuff. 

We use HPC resources via the National Computional Instructure (NCI) in which our clusters use Rocky Linux 8, which uses glib2.28. 

I've been trying to use the following commands to place the 5.1.0. into a data library path (within a bash script):

### Main installation steps ###

wget https://github.com/SuperElastix/elastix/archive/refs/tags/5.1.0.tar.gz
tar -xf 5.1.0.tar.gz -C ${DATA_LIB_PATH}/elastix
chmod +R 777 ${DATA_LIB_PATH}/${APP_NAME}

### End of Main Installation Steps  ###

### create Tool Command Language (TCL) module file  ###
mkdir -p ${APP_MODULE_PATH}
cat<<EOF > ${APP_MODULE_PATH}/${APP_VERSION}
#%Module1.0
source /opt/Modules/extensions/extensions.tcl
prepend-path    LD_LIBRARY_PATH   ${DATA_LIB_PATH}/.local/bin
prepend-path    LD_LIBRARY_PATH   ${DATA_LIB_PATH}/${APP_NAME}/lib
prepend-path    PATH              ${DATA_LIB_PATH}/${APP_NAME}/bin
EOF


However, when I call the executable I'm getting the following errors: 
![image](https://github.com/SuperElastix/elastix/assets/41981766/99d43785-cc70-4355-9006-8f01d9b0b42e)

I can load a C++ compiler, which fixes the GLIBCXX_3.4.26 error but our glibc is a system library that can't be replaced. 

I've been hesistant to build from source as it might not be possible to build with glibc 2.28 (either for elastix or ITK) and I would like some advice.

Kind Regards, 
Kyle

**Comment by N-Dekker** [comment]
Thanks for de details @dorkylever 

> `wget https://github.com/SuperElastix/elastix/archive/refs/tags/5.1.0.tar.gz`

If you do not build elastix from source, 5.1.0.tar.gz may be the wrong file for you, because it just contains the source! Can you possibly still try [elastix-5.1.0-linux.zip](https://github.com/SuperElastix/elastix/releases/download/5.1.0/elastix-5.1.0-linux.zip) from  https://github.com/SuperElastix/elastix/releases/tag/5.1.0 ?

When you then run elastix, I would suggest specifying its file path, for example:

    $ ./your/path/to/elastix

Instead of just

    $ elastix

If that still produces the same error, maybe this information (suggested by my colleague Baldur) could be of help: https://www.baeldung.com/linux/rpath-change-in-binary

**Comment by dorkylever** [comment]
Hi All, 

It ended up being possible to build from source (using gcc 13.2.0).

I originally tried the zip and it still gave the same error. When I checked, the paths in binary were correct and the libraries were linked. It was just that versions of GLIBC are required in the binary (shown below) (GLIBC > 2.25) are not present on the system.  

Can I asked why you switched to building with a GNU compiler?  Compared to gcc, it seems a lot heavier on libc  than c++, which at least for us, is harder to change? 

[kd0793@gadi-login-01 5.1.0]$ readelf -V bin/elastix

Version symbols section '.gnu.version' contains 588 entries:
 Addr: 00000000000078e4  Offset: 0x0078e4  Link: 6 (.dynsym)
  000:   0 (*local*)       2 (GLIBC_2.2.5)   3 (GLIBC_2.3.4)   4 (GLIBCXX_3.4.21)
  004:   5 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)   4 (GLIBCXX_3.4.21)
  008:   5 (GLIBC_2.2.5)   2 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   8 (GLIBC_2.2.5)
  00c:   7 (GLIBCXX_3.4)   6 (GLIBC_2.2.5)   9 (CXXABI_1.3.9)   a (GLIBCXX_3.4.15)
  010:   6 (GLIBC_2.2.5)   4 (GLIBCXX_3.4.21)    6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)
  014:   7 (GLIBCXX_3.4)   5 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)
  018:   6 (GLIBC_2.2.5)   b (GLIBC_2.7)     c (CXXABI_1.3)    d (GLIBCXX_3.4.11)
  01c:   4 (GLIBCXX_3.4.21)    6 (GLIBC_2.2.5)   9 (CXXABI_1.3.9)   4 (GLIBCXX_3.4.21)
  020:   7 (GLIBCXX_3.4)   0 (*local*)       6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)
  024:   e (GLIBCXX_3.4.20)    6 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)   f (GLIBCXX_3.4.9)
  028:  10 (GLIBC_2.29)   10 (GLIBC_2.29)    6 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)
  02c:  11 (GLIBC_2.28)   12 (GLIBC_2.3)     7 (GLIBCXX_3.4)   6 (GLIBC_2.2.5)
  030:   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   b (GLIBC_2.7)     2 (GLIBC_2.2.5)
  034:   7 (GLIBCXX_3.4)   c (CXXABI_1.3)    6 (GLIBC_2.2.5)   0 (*local*)
  038:   f (GLIBCXX_3.4.9)   4 (GLIBCXX_3.4.21)    7 (GLIBCXX_3.4)   6 (GLIBC_2.2.5)
  03c:   6 (GLIBC_2.2.5)   4 (GLIBCXX_3.4.21)    a (GLIBCXX_3.4.15)    4 (GLIBCXX_3.4.21)
  040:   3 (GLIBC_2.3.4)   4 (GLIBCXX_3.4.21)    6 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)
  044:   7 (GLIBCXX_3.4)   6 (GLIBC_2.2.5)   f (GLIBCXX_3.4.9)   6 (GLIBC_2.2.5)
  048:   2 (GLIBC_2.2.5)   4 (GLIBCXX_3.4.21)    5 (GLIBC_2.2.5)   4 (GLIBCXX_3.4.21)
  04c:   6 (GLIBC_2.2.5)  13 (GLIBC_2.4)     5 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)
  050:   d (GLIBCXX_3.4.11)    6 (GLIBC_2.2.5)   8 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)
  054:   6 (GLIBC_2.2.5)   5 (GLIBC_2.2.5)   0 (*local*)       d (GLIBCXX_3.4.11)
  058:  14 (CXXABI_1.3.3)   7 (GLIBCXX_3.4)  15 (CXXABI_1.3.7)  16 (GLIBC_2.3.2)
  05c:   5 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)   7 (GLIBCXX_3.4)   2 (GLIBC_2.2.5)
  060:  17 (GLIBC_2.17)    7 (GLIBCXX_3.4)   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)
  064:   2 (GLIBC_2.2.5)  14 (CXXABI_1.3.3)   c (CXXABI_1.3)    6 (GLIBC_2.2.5)
  068:   0 (*local*)       c (CXXABI_1.3)    5 (GLIBC_2.2.5)   f (GLIBCXX_3.4.9)
  06c:   6 (GLIBC_2.2.5)  18 (GLIBCXX_3.4.22)    d (GLIBCXX_3.4.11)   12 (GLIBC_2.3)
  070:   d (GLIBCXX_3.4.11)    6 (GLIBC_2.2.5)  19 (GLIBCXX_3.4.26)    c (CXXABI_1.3)
  074:   4 (GLIBCXX_3.4.21)    5 (GLIBC_2.2.5)  14 (CXXABI_1.3.3)   6 (GLIBC_2.2.5)
  078:   f (GLIBCXX_3.4.9)   d (GLIBCXX_3.4.11)    a (GLIBCXX_3.4.15)   1a (GCC_4.0.0)
  07c:   f (GLIBCXX_3.4.9)   4 (GLIBCXX_3.4.21)    3 (GLIBC_2.3.4)   7 (GLIBCXX_3.4)
  080:   7 (GLIBCXX_3.4)  14 (CXXABI_1.3.3)   4 (GLIBCXX_3.4.21)    6 (GLIBC_2.2.5)
  084:   4 (GLIBCXX_3.4.21)    7 (GLIBCXX_3.4)   5 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)
  088:   5 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   4 (GLIBCXX_3.4.21)   12 (GLIBC_2.3)
  08c:   6 (GLIBC_2.2.5)   4 (GLIBCXX_3.4.21)    d (GLIBCXX_3.4.11)    7 (GLIBCXX_3.4)
  090:   6 (GLIBC_2.2.5)   4 (GLIBCXX_3.4.21)    c (CXXABI_1.3)    4 (GLIBCXX_3.4.21)
  094:   3 (GLIBC_2.3.4)   7 (GLIBCXX_3.4)   4 (GLIBCXX_3.4.21)    7 (GLIBCXX_3.4)
  098:   7 (GLIBCXX_3.4)  1b (GCC_3.0)       6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)
  09c:   6 (GLIBC_2.2.5)   0 (*local*)       7 (GLIBCXX_3.4)   4 (GLIBCXX_3.4.21)
  0a0:   3 (GLIBC_2.3.4)   0 (*local*)       6 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)
  0a4:  16 (GLIBC_2.3.2)   7 (GLIBCXX_3.4)  1c (GLIBCXX_3.4.14)    6 (GLIBC_2.2.5)
  0a8:   4 (GLIBCXX_3.4.21)    6 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)   4 (GLIBCXX_3.4.21)
  0ac:   2 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)   6 (GLIBC_2.2.5)   4 (GLIBCXX_3.4.21)
  0b0:   f (GLIBCXX_3.4.9)   6 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)   f (GLIBCXX_3.4.9)
  0b4:   3 (GLIBC_2.3.4)   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)
  0b8:   6 (GLIBC_2.2.5)  1d (GLIBC_2.27)    6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)
  0bc:  18 (GLIBCXX_3.4.22)    0 (*local*)       6 (GLIBC_2.2.5)   d (GLIBCXX_3.4.11)
  0c0:   5 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)   c (CXXABI_1.3)    4 (GLIBCXX_3.4.21)
  0c4:  14 (CXXABI_1.3.3)   4 (GLIBCXX_3.4.21)   1e (GLIBCXX_3.4.17)    6 (GLIBC_2.2.5)
  0c8:   6 (GLIBC_2.2.5)   5 (GLIBC_2.2.5)   2 (GLIBC_2.2.5)  1f (OMP_1.0)
  0cc:   6 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)   7 (GLIBCXX_3.4)   6 (GLIBC_2.2.5)
  0d0:   5 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   2 (GLIBC_2.2.5)   3 (GLIBC_2.3.4)
  0d4:   c (CXXABI_1.3)    4 (GLIBCXX_3.4.21)    7 (GLIBCXX_3.4)   2 (GLIBC_2.2.5)
  0d8:   5 (GLIBC_2.2.5)  14 (CXXABI_1.3.3)   7 (GLIBCXX_3.4)   7 (GLIBCXX_3.4)
  0dc:   7 (GLIBCXX_3.4)   6 (GLIBC_2.2.5)  14 (CXXABI_1.3.3)   c (CXXABI_1.3)
  0e0:  19 (GLIBCXX_3.4.26)    d (GLIBCXX_3.4.11)    7 (GLIBCXX_3.4)  20 (GLIBCXX_3.4.5)
  0e4:   7 (GLIBCXX_3.4)   6 (GLIBC_2.2.5)  18 (GLIBCXX_3.4.22)    5 (GLIBC_2.2.5)
  0e8:   6 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)  1c (GLIBCXX_3.4.14)    3 (GLIBC_2.3.4)
  0ec:   7 (GLIBCXX_3.4)   6 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)   5 (GLIBC_2.2.5)
  0f0:   4 (GLIBCXX_3.4.21)    7 (GLIBCXX_3.4)   4 (GLIBCXX_3.4.21)    f (GLIBCXX_3.4.9)
  0f4:   0 (*local*)       5 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)   6 (GLIBC_2.2.5)
  0f8:   6 (GLIBC_2.2.5)  1c (GLIBCXX_3.4.14)    7 (GLIBCXX_3.4)   6 (GLIBC_2.2.5)
  0fc:   5 (GLIBC_2.2.5)  1c (GLIBCXX_3.4.14)    5 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)
  100:   6 (GLIBC_2.2.5)   a (GLIBCXX_3.4.15)    6 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)
  104:   5 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)
  108:   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   4 (GLIBCXX_3.4.21)
  10c:   4 (GLIBCXX_3.4.21)    6 (GLIBC_2.2.5)   f (GLIBCXX_3.4.9)   5 (GLIBC_2.2.5)
  110:   8 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)
  114:   7 (GLIBCXX_3.4)  21 (GLIBC_2.14)   22 (CXXABI_1.3.2)   7 (GLIBCXX_3.4)
  118:   4 (GLIBCXX_3.4.21)    4 (GLIBCXX_3.4.21)    5 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)
  11c:   7 (GLIBCXX_3.4)   6 (GLIBC_2.2.5)   3 (GLIBC_2.3.4)   6 (GLIBC_2.2.5)
  120:   f (GLIBCXX_3.4.9)   0 (*local*)       f (GLIBCXX_3.4.9)   6 (GLIBC_2.2.5)
  124:   7 (GLIBCXX_3.4)   7 (GLIBCXX_3.4)   4 (GLIBCXX_3.4.21)    a (GLIBCXX_3.4.15)
  128:   7 (GLIBCXX_3.4)   4 (GLIBCXX_3.4.21)    6 (GLIBC_2.2.5)   2 (GLIBC_2.2.5)
  12c:   3 (GLIBC_2.3.4)   6 (GLIBC_2.2.5)   4 (GLIBCXX_3.4.21)    6 (GLIBC_2.2.5)
  130:   c (CXXABI_1.3)    5 (GLIBC_2.2.5)   2 (GLIBC_2.2.5)   4 (GLIBCXX_3.4.21)
  134:   7 (GLIBCXX_3.4)   7 (GLIBCXX_3.4)   a (GLIBCXX_3.4.15)    c (CXXABI_1.3)
  138:   7 (GLIBCXX_3.4)   5 (GLIBC_2.2.5)   4 (GLIBCXX_3.4.21)    2 (GLIBC_2.2.5)
  13c:   c (CXXABI_1.3)   10 (GLIBC_2.29)    5 (GLIBC_2.2.5)   4 (GLIBCXX_3.4.21)
  140:   7 (GLIBCXX_3.4)   c (CXXABI_1.3)    6 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)
  144:  12 (GLIBC_2.3)     6 (GLIBC_2.2.5)   0 (*local*)       f (GLIBCXX_3.4.9)
  148:   6 (GLIBC_2.2.5)  1d (GLIBC_2.27)   14 (CXXABI_1.3.3)   6 (GLIBC_2.2.5)
  14c:   4 (GLIBCXX_3.4.21)    f (GLIBCXX_3.4.9)   c (CXXABI_1.3)   16 (GLIBC_2.3.2)
  150:   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)   6 (GLIBC_2.2.5)
  154:   2 (GLIBC_2.2.5)   4 (GLIBCXX_3.4.21)    c (CXXABI_1.3)    7 (GLIBCXX_3.4)
  158:   6 (GLIBC_2.2.5)   4 (GLIBCXX_3.4.21)   1a (GCC_4.0.0)     8 (GLIBC_2.2.5)
  15c:  24 (GLIBC_2.8)     c (CXXABI_1.3)   25 (CXXABI_1.3.11)   7 (GLIBCXX_3.4)
  160:   7 (GLIBCXX_3.4)   4 (GLIBCXX_3.4.21)    7 (GLIBCXX_3.4)  26 (GLIBC_2.3)
  164:   7 (GLIBCXX_3.4)   7 (GLIBCXX_3.4)   6 (GLIBC_2.2.5)   5 (GLIBC_2.2.5)
  168:   7 (GLIBCXX_3.4)   4 (GLIBCXX_3.4.21)    6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)
  16c:   5 (GLIBC_2.2.5)   c (CXXABI_1.3)   27 (GLIBC_2.15)    7 (GLIBCXX_3.4)
  170:  1a (GCC_4.0.0)     4 (GLIBCXX_3.4.21)   16 (GLIBC_2.3.2)   6 (GLIBC_2.2.5)
  174:   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   4 (GLIBCXX_3.4.21)
  178:   7 (GLIBCXX_3.4)   6 (GLIBC_2.2.5)   4 (GLIBCXX_3.4.21)    6 (GLIBC_2.2.5)
  17c:   5 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)   7 (GLIBCXX_3.4)   c (CXXABI_1.3)
  180:   6 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)   6 (GLIBC_2.2.5)  1a (GCC_4.0.0)
  184:   d (GLIBCXX_3.4.11)    6 (GLIBC_2.2.5)   5 (GLIBC_2.2.5)   5 (GLIBC_2.2.5)
  188:   a (GLIBCXX_3.4.15)    c (CXXABI_1.3)    6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)
  18c:   6 (GLIBC_2.2.5)  12 (GLIBC_2.3)     f (GLIBCXX_3.4.9)   6 (GLIBC_2.2.5)
  190:   7 (GLIBCXX_3.4)   7 (GLIBCXX_3.4)   6 (GLIBC_2.2.5)   5 (GLIBC_2.2.5)
  194:   5 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)   5 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)
  198:  28 (GLIBC_2.3.2)  29 (GLIBC_2.6)     d (GLIBCXX_3.4.11)    4 (GLIBCXX_3.4.21)
  19c:   4 (GLIBCXX_3.4.21)    f (GLIBCXX_3.4.9)   7 (GLIBCXX_3.4)   6 (GLIBC_2.2.5)
  1a0:   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   f (GLIBCXX_3.4.9)   6 (GLIBC_2.2.5)
  1a4:   f (GLIBCXX_3.4.9)   6 (GLIBC_2.2.5)  2a (GLIBCXX_3.4.19)    6 (GLIBC_2.2.5)
  1a8:   2 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)
  1ac:   7 (GLIBCXX_3.4)   5 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)
  1b0:   d (GLIBCXX_3.4.11)    7 (GLIBCXX_3.4)  20 (GLIBCXX_3.4.5)   c (CXXABI_1.3)
  1b4:   4 (GLIBCXX_3.4.21)    4 (GLIBCXX_3.4.21)    6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)
  1b8:   7 (GLIBCXX_3.4)   c (CXXABI_1.3)    3 (GLIBC_2.3.4)   6 (GLIBC_2.2.5)
  1bc:   5 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)
  1c0:   c (CXXABI_1.3)    7 (GLIBCXX_3.4)   8 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)
  1c4:   4 (GLIBCXX_3.4.21)    4 (GLIBCXX_3.4.21)    4 (GLIBCXX_3.4.21)    5 (GLIBC_2.2.5)
  1c8:   7 (GLIBCXX_3.4)   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)
  1cc:  2b (CXXABI_1.3.8)   4 (GLIBCXX_3.4.21)    6 (GLIBC_2.2.5)   c (CXXABI_1.3)
  1d0:   2 (GLIBC_2.2.5)   c (CXXABI_1.3)    6 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)
  1d4:  2c (GLIBC_2.25)    5 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)  1d (GLIBC_2.27)
  1d8:   c (CXXABI_1.3)    6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)
  1dc:   4 (GLIBCXX_3.4.21)    6 (GLIBC_2.2.5)   4 (GLIBCXX_3.4.21)    4 (GLIBCXX_3.4.21)
  1e0:   4 (GLIBCXX_3.4.21)   12 (GLIBC_2.3)     5 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)
  1e4:   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   a (GLIBCXX_3.4.15)
  1e8:   c (CXXABI_1.3)    5 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)  13 (GLIBC_2.4)
  1ec:   7 (GLIBCXX_3.4)   2 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)
  1f0:   5 (GLIBC_2.2.5)   5 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)   6 (GLIBC_2.2.5)
  1f4:   3 (GLIBC_2.3.4)   a (GLIBCXX_3.4.15)    d (GLIBCXX_3.4.11)    6 (GLIBC_2.2.5)
  1f8:   2 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)   7 (GLIBCXX_3.4)   7 (GLIBCXX_3.4)
  1fc:   3 (GLIBC_2.3.4)  1c (GLIBCXX_3.4.14)    6 (GLIBC_2.2.5)   4 (GLIBCXX_3.4.21)
  200:   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   4 (GLIBCXX_3.4.21)    4 (GLIBCXX_3.4.21)
  204:   6 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)   6 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)
  208:   7 (GLIBCXX_3.4)  25 (CXXABI_1.3.11)   6 (GLIBC_2.2.5)   5 (GLIBC_2.2.5)
  20c:   7 (GLIBCXX_3.4)   c (CXXABI_1.3)    6 (GLIBC_2.2.5)   d (GLIBCXX_3.4.11)
  210:   4 (GLIBCXX_3.4.21)    6 (GLIBC_2.2.5)   3 (GLIBC_2.3.4)   6 (GLIBC_2.2.5)
  214:   2 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)   1 (*global*)
  218:   1 (*global*)      1 (*global*)      c (CXXABI_1.3)    1 (*global*)
  21c:   1 (*global*)      7 (GLIBCXX_3.4)   1 (*global*)      7 (GLIBCXX_3.4)
  220:   7 (GLIBCXX_3.4)   6 (GLIBC_2.2.5)   6 (GLIBC_2.2.5)   7 (GLIBCXX_3.4)
  224:   7 (GLIBCXX_3.4)   a (GLIBCXX_3.4.15)    1 (*global*)      7 (GLIBCXX_3.4)
  228:   1 (*global*)      7 (GLIBCXX_3.4)   4 (GLIBCXX_3.4.21)    7 (GLIBCXX_3.4)
  22c:   7 (GLIBCXX_3.4)   7 (GLIBCXX_3.4)   6 (GLIBC_2.2.5)   c (CXXABI_1.3)
  230:   4 (GLIBCXX_3.4.21)    7 (GLIBCXX_3.4)   7 (GLIBCXX_3.4)   a (GLIBCXX_3.4.15)
  234:   1 (*global*)      6 (GLIBC_2.2.5)   c (CXXABI_1.3)    6 (GLIBC_2.2.5)
  238:   c (CXXABI_1.3)    7 (GLIBCXX_3.4)   0 (*local*)       4 (GLIBCXX_3.4.21)
  23c:   c (CXXABI_1.3)    7 (GLIBCXX_3.4)   4 (GLIBCXX_3.4.21)    c (CXXABI_1.3)
  240:   4 (GLIBCXX_3.4.21)    c (CXXABI_1.3)    4 (GLIBCXX_3.4.21)    7 (GLIBCXX_3.4)
  244:   c (CXXABI_1.3)    4 (GLIBCXX_3.4.21)    1 (*global*)      1 (*global*)
  248:   c (CXXABI_1.3)   23 (GLIBC_2.11)    1 (*global*)      7 (GLIBCXX_3.4)

Version needs section '.gnu.version_r' contains 8 entries:
 Addr: 0x0000000000007d80  Offset: 0x007d80  Link: 7 (.dynstr)
  000000: Version: 1  File: ld-linux-x86-64.so.2  Cnt: 1
  0x0010:   Name: GLIBC_2.3  Flags: none  Version: 38
  0x0020: Version: 1  File: libgomp.so.1  Cnt: 1
  0x0030:   Name: OMP_1.0  Flags: none  Version: 31
  0x0040: Version: 1  File: libgcc_s.so.1  Cnt: 2
  0x0050:   Name: GCC_3.0  Flags: none  Version: 27
  0x0060:   Name: GCC_4.0.0  Flags: none  Version: 26
  0x0070: Version: 1  File: libdl.so.2  Cnt: 1
  0x0080:   Name: GLIBC_2.2.5  Flags: none  Version: 8
  0x0090: Version: 1  File: libm.so.6  Cnt: 3
  0x00a0:   Name: GLIBC_2.27  Flags: none  Version: 29
  0x00b0:   Name: GLIBC_2.29  Flags: none  Version: 16
  0x00c0:   Name: GLIBC_2.2.5  Flags: none  Version: 5
  0x00d0: Version: 1  File: libstdc++.so.6  Cnt: 19
  0x00e0:   Name: CXXABI_1.3.8  Flags: none  Version: 43
  0x00f0:   Name: GLIBCXX_3.4.19  Flags: none  Version: 42
  0x0100:   Name: CXXABI_1.3.11  Flags: none  Version: 37
  0x0110:   Name: CXXABI_1.3.2  Flags: none  Version: 34
  0x0120:   Name: GLIBCXX_3.4.5  Flags: none  Version: 32
  0x0130:   Name: GLIBCXX_3.4.17  Flags: none  Version: 30
  0x0140:   Name: GLIBCXX_3.4.14  Flags: none  Version: 28
  0x0150:   Name: GLIBCXX_3.4.26  Flags: none  Version: 25
  0x0160:   Name: GLIBCXX_3.4.22  Flags: none  Version: 24
  0x0170:   Name: CXXABI_1.3.7  Flags: none  Version: 21
  0x0180:   Name: CXXABI_1.3.3  Flags: none  Version: 20
  0x0190:   Name: GLIBCXX_3.4.9  Flags: none  Version: 15
  0x01a0:   Name: GLIBCXX_3.4.20  Flags: none  Version: 14
  0x01b0:   Name: GLIBCXX_3.4.11  Flags: none  Version: 13
  0x01c0:   Name: CXXABI_1.3  Flags: none  Version: 12
  0x01d0:   Name: GLIBCXX_3.4.15  Flags: none  Version: 10
  0x01e0:   Name: CXXABI_1.3.9  Flags: none  Version: 9
  0x01f0:   Name: GLIBCXX_3.4  Flags: none  Version: 7
  0x0200:   Name: GLIBCXX_3.4.21  Flags: none  Version: 4
  0x0210: Version: 1  File: libc.so.6  Cnt: 14
  0x0220:   Name: GLIBC_2.25  Flags: none  Version: 44
  0x0230:   Name: GLIBC_2.6  Flags: none  Version: 41
  0x0240:   Name: GLIBC_2.3.2  Flags: none  Version: 40
  0x0250:   Name: GLIBC_2.15  Flags: none  Version: 39
  0x0260:   Name: GLIBC_2.8  Flags: none  Version: 36
  0x0270:   Name: GLIBC_2.11  Flags: none  Version: 35
  0x0280:   Name: GLIBC_2.14  Flags: none  Version: 33
  0x0290:   Name: GLIBC_2.17  Flags: none  Version: 23
  0x02a0:   Name: GLIBC_2.4  Flags: none  Version: 19
  0x02b0:   Name: GLIBC_2.3  Flags: none  Version: 18
  0x02c0:   Name: GLIBC_2.28  Flags: none  Version: 17
  0x02d0:   Name: GLIBC_2.7  Flags: none  Version: 11
  0x02e0:   Name: GLIBC_2.2.5  Flags: none  Version: 6
  0x02f0:   Name: GLIBC_2.3.4  Flags: none  Version: 3
  0x0300: Version: 1  File: libpthread.so.0  Cnt: 2
  0x0310:   Name: GLIBC_2.3.2  Flags: none  Version: 22
  0x0320:   Name: GLIBC_2.2.5  Flags: none  Version: 2

**Comment by N-Dekker** [comment]
> Can I asked why you switched to building with a GNU compiler? Compared to gcc, it seems a lot heavier on libc than c++, which at least for us, is harder to change?

Thanks for asking. However, I don't really understand the question 🤷 GCC = The GNU C++ compiler! (Or more precisely: _the GNU Compiler Collection_: https://www.gnu.org/software/gcc )  So we do use GCC to build elastix binaries for Linux, we did not "switch"!
